### ISSUE TITLE
MergeBamAlignment does not support streaming input anymore

### ISSUE THREAD

**Issue by nh13** [broadinstitute/picard#734]
@yfarjoun seems like the latest commit introduced a bug: https://github.com/broadinstitute/picard/commit/afb4af37be4a38eb94b50a09ed6bf275de151472
Also, the commit subject has nothing to do with the rest of the message as far as I can tell.  

Can you take a look?  Can we revert master if you think it'll take a while?  Please add some tests too.

**Comment by nh13** [comment]
I forgot the stacktrace:
```
Exception in thread "main" htsjdk.samtools.SAMException: Cannot figure out type of file /dev/stdin from extension. Current implementation understands the following types: [FASTA: [.fa.gz, .fna.gz, .txt, .fasta.gz, .fa, .fasta, .fna, .txt.gz], DICTIONARY: [.dict], SAM: [.bam, .sam], VCF: [.bcf, .vcf.gz, .vcf], INTERVAL_LIST: [.interval_list]]
        at htsjdk.variant.utils.SAMSequenceDictionaryExtractor$TYPE.forFile(SAMSequenceDictionaryExtractor.java:124)
        at htsjdk.variant.utils.SAMSequenceDictionaryExtractor.extractDictionary(SAMSequenceDictionaryExtractor.java:134)
        at picard.sam.SamAlignmentMerger.getDictionaryForMergedBam(SamAlignmentMerger.java:193)
        at picard.sam.AbstractAlignmentMerger.mergeAlignment(AbstractAlignmentMerger.java:336)
        at picard.sam.SamAlignmentMerger.mergeAlignment(SamAlignmentMerger.java:180)
        at picard.sam.MergeBamAlignment.doWork(MergeBamAlignment.java:283)
        at picard.cmdline.CommandLineProgram.instanceMain(CommandLineProgram.java:205)
        at picard.cmdline.PicardCommandLine.instanceMain(PicardCommandLine.java:94)
        at picard.cmdline.PicardCommandLine.main(PicardCommandLine.java:104)
```

**Comment by tfenne** [comment]
Note that simply fixing the SequenceDictionaryExtractor to recognize /dev/stdin won't work, the problem will just get pushed a little further down, because we simply can't read the input aligned file multiple times.

**Comment by yfarjoun** [comment]
hmmm. sorry about that. I'll revert and fix.

**Comment by yfarjoun** [comment]
fixed by #739
